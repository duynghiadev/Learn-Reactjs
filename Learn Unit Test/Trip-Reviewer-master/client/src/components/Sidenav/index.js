import React from 'react';
import PropTypes from 'prop-types';
import ReactSideNav from 'react-simple-sidenav';

import {
  home,
  userProfile,
  signup,
  login,
  userReviews,
  addTrip,
  logout,
} from '../../constants/navigation-items';
import SidenavItem from '../../containers/NavigationItem';

const order = [home, userProfile, signup, login, addTrip, userReviews, logout];

const Sidenav = props => (
  <ReactSideNav
    showNav={props.showNav}
    onHideNav={props.onHideNav}
    navStyle={{
      background: '#242424',
      maxWidth: '220px',
    }}
  >
    {order.map(item => (
      <div key={item} onClick={props.onHideNav}>
        <SidenavItem item={item} sidenav />
      </div>
    ))}
  </ReactSideNav>
);

Sidenav.propTypes = {
  showNav: PropTypes.bool.isRequired,
  onHideNav: PropTypes.func.isRequired,
};

export default Sidenav;
