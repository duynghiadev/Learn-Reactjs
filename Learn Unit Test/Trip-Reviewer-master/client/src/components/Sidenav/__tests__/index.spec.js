import React from 'react';
import { create } from 'react-test-renderer';

import Sidenav from '..';

jest.mock('react-simple-sidenav', () => 'ReactSideNav');
jest.mock('./../../../containers/NavigationItem', () => 'NavigationItem');

const mockComponent = props => <Sidenav {...props} />;

describe('<Sidenav />', () => {
  it('should render component', () => {
    const props = {
      onHideNav: () => {},
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
