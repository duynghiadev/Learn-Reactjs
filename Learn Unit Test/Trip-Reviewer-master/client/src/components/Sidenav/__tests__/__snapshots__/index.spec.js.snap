// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidenav /> should render component 1`] = `
<ReactSideNav
  navStyle={
    Object {
      "background": "#242424",
      "maxWidth": "220px",
    }
  }
  onHideNav={[Function]}
  showNav={undefined}
>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="HOME"
      sidenav={true}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="USER_PROFILE"
      sidenav={true}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="SIGNUP"
      sidenav={true}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="LOGIN"
      sidenav={true}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="ADD_TRIP"
      sidenav={true}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="USER_REVIEWS"
      sidenav={true}
    />
  </div>
  <div
    onClick={[Function]}
  >
    <NavigationItem
      item="LOGOUT"
      sidenav={true}
    />
  </div>
</ReactSideNav>
`;
